tests/test_mock_sampler.py

```diff
@@ -216,7 +217,90 @@ def test_yield_arguments(self):
                                    broken_edges=delete_edges)
         self.assertTrue(len(sampler.nodelist)==4)
         self.assertTrue(len(sampler.edgelist)==1)
-        
+
+    def test_custom_substitute_sampler(self):
+        """Test that MockDWaveSampler uses the provided custom substitute_sampler."""
+
+        # Define a sampler that always returns the a constant (excited) state
+        class SteepestAscentSolver(SteepestDescentSolver):
+            def __init__(self):
+                super().__init__()
+
+            def sample(self, bqm, **kwargs):
+                # Return local (or global)  maxima instead of local minima
```

Comment:
```suggestion
                # Return local (or global)  maxima instead of local minima
                # NOTE: energy returned is not faithful to the original bqm (energy calculated as `-bqm`)
```